### DISCUSSION TITLE
Biff's Bolognese (3h)

### DISCUSSION THREAD

**Discussion by richardbirkin** [SuperHyperInstantFutureTime/TrackShift#25] [Meetings]
2.5h @richardbirkin @g105b 
0.5h @richardbirkin write-up

## SLC not MVP

Greg told me about using SLC instead of MVP. It stands for '[Simple, Loveable, Complete](https://github.com/SuperHyperInstantFutureTime/Trackshift/discussions/25#discussioncomment-4157011)'
It's better than 'MVP' as _minimum viable_ implies that it's a poorer version of something else. What Greg has made so far is more than that in that it satisfies a core user requirement, and does it completely, without fuss. And I love it. 

Greg also had a good side rant about _ergonomic coding_. When I hear the word 'ergonomic' I think of the Ergonomes we made in technology class at school, in order to design the best chair. Well, chairs are easy to sit in and comfortable because of the hard work that has gone into the design. I think that's what Greg was getting at - the code has been designed to be as simple and effective as possible.

## TrackShift SLC v1

A simple PRS Distribution Statement organiser for Artists. 

What's next?

Deploy it to a server and, less importantly, get it working locally on my machine (and document this). 

## Github Milestones + Release

We've done Planning into Issues into Actions. Now it's time to frame that work into Milestones and Releases. 

### Milestone 1

Multi-PRS Statement Processor for Artists. 

- [x] Upload a CSV
- [x] Upload 10 CSVs
- [x] Overview of Tunes and Amounts

### Milestone 2

Manage uploads + data policy

- [x] View uploads
- [x] Edit uploads - Remove CSV(s)
- [x] Privacy Policy + Explanation re: once cookie is gone, data is gone and files deleted
- [x] Countdown to deletion of data
- [x] Extend time until deletion of data

### Milestone 3

- [x] Set up Deployment
- [x] Handle multiple files uploaded at once
- [x] Basic Drag & Drop interface

### Milestone 4

- [x] Add minimal CSS like [Better Motherfucking Website](http://bettermotherfuckingwebsite.com/)

### Milestone 5

TBD. Maybe Multi-statement type processor for Labels.

This will need breaking down into tasks and planning properly.

## Rounding Up Or Down

I need to check what happens in PRS statements. Greg is currently rounding down to two decimal places, which makes sense as by rounding up you're making up money. But needs double checking. 

## Being Punk About Data

We discussed generating randomised data sets to use in tests instead of actual files, since they would have to be stored in the project. The only reason we're even looking at real statements is to get the format and header rows. 

This then branched into a discussion about Milestone 2. We're not adding AUTH yet. We feel it would change the product and make it less SLC. We're not even using a database yet.

We're storing the user's files in folders created for the user's session. The session will expire, and the data will be deleted. The user should be able to extend this time if the product is adding value. Or they could upload the files again, since it's so Simple. 

Value as a one-way street. We don't want your data in return for giving you value. This is a passion project. There will be no selling of data. No bads. No ads. \m/

## Note to Greg

Write upload 'terms' from code. 
e.g. File size limit. Storage time etc. 

This should also be applied to the Privacy Policy - it should be generated from a test-driven class. 

## Cattle Not Pets

Greg introduced me to the concept of servers being cattle, not pets. Don't name them, and cull when done. A totally fresh droplet for a new deployment, with detached resources and assets. Test the new one, then delete the old one.

### Funding 

We discussed what is next for me following my introduction to Grantify. 

We need to reinstate our company so that we have a company to apply for a Smart Grant with. 

Greg has experience on projects that have been successful. His advice is to ask for people's help. Go to universities and find the people who can help, and the people who know the Innovate UK people. 

Get pointed in the right direction by talking to knowledgable people about the product. 

### Where do we sit in the business?

We've spoken about this before but never written it down. 

We are co-founders, and own 50% each. Our combined shares will never be diluted less that 51%. 

I am Managing/Creative Director. 

Greg is Technical Director. 

### How to Spend Half A Million Quid

Our contact at Grantify suggested we apply for £500K. We sketched how we might use that, as it sounds like a lot of money but from my experience in startups you can burn through that very quickly. 

So...our napkin calcs for a 1 year 'runway':

£200K combined full-time salaries + overheads
£150K legal/financial consultants
£50K travel to make partnerships, set up board, network.
£50K design/dev freelance
£50K marketing

## Viewing Tests

We broke the meeting rule and opened a laptop, but only so that Greg can show me Github running the tests and explain how the code works. Bloody brilliant.

**Comment by g105b** [comment]
> **Simple**
> When you build a Simple, Lovable and Complete product the focus is on creating a simple application that does its job very well, not a minimal one that just does ok. This applies to both the functionality of the product and the UX design. Most customers don’t need lots of bells and whistles - but they do want a great user experience.
> 
> **Lovable**
> Because SLC products are designed with the customers emotions in mind, people actually enjoy using them right from the start. The product provides real value, not just minimum viability. If done well, it will even surprise and delight. This makes it much easier for customers to develop a genuine affinity for the new product, which can also extend to brand loyalty toward the company.
> 
> **Complete**
> Though it may be small at first, an SLC application is still built to be a complete product. Customer needs are thoroughly understood and optimized for right from the start, and the product is designed to provide the most important functionality, not just the most basic. Genuine value is created without any further development.
